scripts/content-repo-preview/build.ts

```diff
@@ -0,0 +1,95 @@
+import { execFileSync } from 'child_process'
+import path from 'path'
+import fs from 'fs'
+
+async function main() {
+  // TODO: check for necessary env vars?
+
+  const repo = process.env.REPO
+
+  // our CWD
+  const cwd = process.cwd()
+  const globalCSSFile = path.join(cwd, 'src', 'pages', 'style.css')
+
+  /**
+   * Check for a .next folder, if found copy it back into our website-preview dir
+   * This should allow us to take advantage of Vercel's build cache
+   */
+  if (fs.existsSync(path.join(cwd, '..', '.next'))) {
+    console.log('.next folder found, moving into website-preview...')
+    execFileSync('mv', ['../.next', './.next'])
+
+    if (fs.existsSync(path.join(cwd, '.next', 'cache', 'node_modules'))) {
+      console.log('Found cached node_modules, moving...')
+      execFileSync('mv', ['./.next/cache/node_modules', './node_modules'])
+    }
+  }
+
+  // copy public files
+  console.log('📝 copying files in the public folder')
+  execFileSync('cp', ['-R', './public', '../'])
+
+  /**
+   * exclude any imports in the global CSS file which rely on other products
+   */
+  const globalCSSFileContents = await fs.promises.readFile(
+    globalCSSFile,
+    'utf-8'
+  )
+
+  const newContents = globalCSSFileContents
```

Comment:
🤩 ✨ nicely done, so much better than the manual specification 💯 also a perfect example of a thing that i would have a very hard time doing in a bash script, but an easy time doing in a node script. Thanks for setting this up as the latter 🙇